DM2C/code/model.py

```diff
@@ -0,0 +1,367 @@
+        self.train_loader_ordered = Custom_Dataloader(dataset=train_data, modal = modalities,
+                                               batch_size=self.args.batch_size,
+                                               shuffle=False)
+
+
+    def embedding(self, dataloader, unify_modal='img'): # actually encodes / makes predictions
+        self.set_model_status(training=False)
+        with torch.no_grad():
+            latent = None
+            for step, (txt_embed, img_embed) in enumerate(dataloader):
+                if unify_modal == 'img':
+                    latent, _ = self.txt2img(txt_embed)
+                elif unify_modal == 'txt':
+                    latent, _ = self.img2txt(img_embed)
+                else:
+                    latent = (txt_embed, img_embed)
+            return latent.cpu().numpy()
+
+
+    # no pretraining needed -- no pretrain dataloader needed
+    def set_model_status(self, training=True):
+        if training:
+            self.img2txt.train()
+            self.txt2img.train()
+            self.D_img.train()
+            self.D_txt.train()
+        else:
+            self.img2txt.eval()
+            self.txt2img.eval()
+            self.D_img.eval()
+            self.D_txt.eval()
+
+    def to_cuda(self):
+        self.img2txt.cuda()
+        self.txt2img.cuda()
+        self.D_img.cuda()
+        self.D_txt.cuda()
+
+    def save_cpt(self, epoch):
+        state_dict = {'epoch': epoch,
+                      'G12_state_dict': self.img2txt.state_dict(),
+                      'G21_state_dict': self.txt2img.state_dict(),
+                      'D1_state_dict': self.D_img.state_dict(),
+                      'D2_state_dict': self.D_txt.state_dict(),
+                      'optimizer_G': self.optimizer_G.state_dict(),
+                      'optimizer_D': self.optimizer_D.state_dict()
+                      }
+        cptname = '{}_checkpt_{}.pkl'.format(self.args.dataset, epoch)
+        cptpath = os.path.join(self.args.cpt_dir, cptname)
+        self.logger.info("> Save checkpoint '{}'".format(cptpath))
+        torch.save(state_dict, cptpath)
+
+    # no need to save pretrain
+    def load_cpt(self, cptpath):
+        if os.path.isfile(cptpath):
+            self.logger.info("> Load checkpoint '{}'".format(cptpath))
+            dicts = torch.load(cptpath)
+            self.epoch = dicts['epoch']
+            self.img2txt.load_state_dict(dicts['G12_state_dict'])
+            self.txt2img.load_state_dict(dicts['G21_state_dict'])
+            self.D_img.load_state_dict(dicts['D1_state_dict'])
+            self.D_txt.load_state_dict(dicts['D2_state_dict'])
+            self.optimizer_G.load_state_dict(dicts['optimizer_G'])
+            self.optimizer_D.load_state_dict(dicts['optimizer_D'])
+            # self.scheduler.load_state_dict(dicts['scheduler'])
+        else:
+            self.logger.error("> No checkpoint found at '{}'".format(cptpath))
+    def set_writer(self):
```

Comment:
Put at least one (or better yet, two) blank lines between separate function and class definitions.